src/glove.c

```diff
@@ -250,13 +250,13 @@ int save_params(int nb_iter) {
                 for (b = 0; b < (vector_size + 1); b++) fprintf(fgs," %lf", gradsq[(vocab_size + a) * (vector_size + 1) + b]);
                 fprintf(fgs,"\n");
             }
-            if (fscanf(fid,format,word) == 0) return 1; // Eat irrelevant frequency entry
+            if (fscanf(fid,format,word) == 0) {free(word); return 1;} // Eat irrelevant frequency entry
         }
 
         if (use_unk_vec) {
             real* unk_vec = (real*)calloc((vector_size + 1), sizeof(real));
             real* unk_context = (real*)calloc((vector_size + 1), sizeof(real));
-            word = "<unk>";
+            strcpy(word, "<unk>");
```

Comment:
Direct assignment
``word = "<unk>";``
makes that word will take the address of string literal. In this case address to a previous memory location (line 188) will be lost and there will be no possibility to free it later.

Reply:
Thank you for looking into this.  This is the first time I have looked into this at all, so please bear with me.

Reply:
Why is this on the heap at all?  Can't it simply be put on the stack?

Reply:
This was my first idea... but I am not sure if it was not done intentionally. 
_save_params_() is called from _train_glove_() where pthreads are created. I can be wrong, but as I know threads share common heap... Is it possible that this was the main reason to allocate it dynamically? Or maybe it was a matter of efficiency?

Reply:
I don't see any benefit to putting it on the heap - my C knowledge is quite out of date, but isn't stack more efficient?  There are fewer instructions needed to get the right address and the allocation is simply "allocate a bigger stack"

Reply:
In terms of fixing things that are wrong this isn't super critical though, not as important as fixing the leak itself